### PR TITLE
[TE] rootcause poc - settings time picker refactor

### PR DESCRIPTION
**Pull request by apucher**
This PR supports time ranges and granularities independent of the primary metric and its maxTime. Also, the display date range picker is now constrained to full days. +Minor UI fixes

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js [hunk 1]
```diff
@@ -58,7 +68,7 @@ export default Ember.Component.extend({
     get() {
       const start = this.get('anomalyRangeStart');
 
-      return start ? moment(+start).format(serverDateFormat) : moment().format(serverDateFormat);
+      return moment(start).format(serverDateFormat);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js]: the `+` here was handling the case where the ms timestamp is a string. just make sure that this doesn't break when you have query params initially

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/template.hbs [hunk 2]
```diff
@@ -13,16 +13,15 @@
...
-            minDate=minDate
-            maxDate=maxDate
+            timePickerIncrement=5
             start=datePickerAnomalyRangeStart
             end=datePickerAnomalyRangeEnd
+            maxDate=datePickerMaxDate
             ranges=anomalyRanges
             showCustomRangeLabel=false
-            format=uiDateFormat
+            format="MMM D, YYYY hh:mm a"
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/template.hbs]: fyi the `timePicker`  `uiDateFormat` here was an ask from Kishore, he wanted the date ranges to depend on the granularity

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js [hunk 3]
```diff
@@ -5,20 +5,20 @@ import moment from 'moment';
...
-
-const testContext = { urns, anomalyRange, compareMode, analysisRange, granularity };
-const testSelectedUrns = new Set([
-  'thirdeye:metric:194591', 'frontend:metric:current:194591', 'frontend:metric:baseline:194591',
-  'thirdeye:metric:194592', 'frontend:metric:current:194592', 'frontend:metric:baseline:194592',
-  'thirdeye:metric:194591:browserName=chrome:browserName=firefox:browserName=safari',
-  'frontend:metric:current:194591:browserName=chrome:browserName=firefox:browserName=safari',
-  'frontend:metric:baseline:194591:browserName=chrome:browserName=firefox:browserName=safari',
-  'thirdeye:event:holiday:2712391']);
+// const anomalyRange = [1509044400000, 1509422400000];
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: let's delete this if we don't need it anymore

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/template.hbs (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/template.hbs]: yes. if it ever comes up, please blame it on me. :-)

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js]: yes, this is guaranteed to be in millis.

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js (hunk 3 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: fixed

**Pull request merged**